system/jlib/jsmartsock.cpp

```diff
@@ -207,6 +208,38 @@ void CSmartSocket::close()
     }
 }
 
+CSmartSocketFactory::CSmartSocketFactory(IPropertyTree &service, bool _retry, unsigned _retryInterval, unsigned _dnsInterval)
+{
+    const char *name = service.queryProp("@name");
+    const char *port = service.queryProp("@port");
+    if (isEmptyString(name) || isEmptyString(port))
+        throw createSmartSocketException(0, "CSmartSocket factory both name and port required for service configuration");
+
+    tlsService  = service.getPropBool("@tls");
+    if (tlsService)
+        tlsConfig.setown(createTlsClientSecretInfo(service.queryProp("@issuer"), service.getPropBool("@pulic")==false, service.getPropBool("@selfSigned"), service.getPropBool("@caCert")));
```

Comment:
trivial (and prev. comment from Gavin): more standard/clearer to use !service.getPropBool("@public")